demo-app/src/open-truss/configs/4-trino-demo.yaml

```diff
@@ -1,9 +1,27 @@
 workflow:
   version: 1
+  signals:
+    results: Records
   frames:
     - frame:
-      data:
-        query: select * from  tpch.sf1.customer limit 10
-        source: trino-demo
       view:
-        component: TrinoDemo
+        component: OTUqiDataProvider
+        props:
+          source: trino-demo
+          query: >
+            SELECT name, phone, acctbal, comment
+            FROM tpch.sf1.customer
+            limit 10
+          results: :results
```

Comment:
One thing that is a bit magical that I'm not the biggest fan of is this `results` prop is not in the components type definition. The component doesn't know what signals the user is going to set ahead of time so we couldn't model it here. Furthermore, what isn't obvious is the name of the key needs to match the name of `results` 2 lines down. For every key in `output` the component looks for a key on `prop` with that name and assumes it's a signal that it can throw the data into. 

This might be more verbose, but one thought is we could do an annotation like this

```yaml
output:
  results:
    signal: :results
    type:
      - name: string
      - phone: string
      - acctbal: number
      - comment: string
```

We would need to update the process props signals code to recursively go through the props and set anything that is a `:symbol` with a signal, but this way would remove the hidden dependency between output and other properties. Additionally, this will allow us to model the `output` prop better in the component's prop interface instead of looking for props that aren't declared by the interface.

Reply:
@kmcq Jon and I were reviewing this and noticed that a consequence of doing this is workflows / components would not be declaring what signals they are using which would make it harder to connect related workflows to each other. Do you think that is the case as well? If not, are there other ways we can know how to connect workflows / components togethers?

Reply:
This is a great concern! I'm realizing that I don't agree with my last sentence. It's best if components declare their signals 👍

Reply:
That said, it does seem like a way for us to determine which components are compatible from reading configs might be really helpful. It would also let us have more structured signals without changing the code. For example let's say we had a query that returned a User and a User is a defined signal:


```ts
interface User {
  id: number
  name: string
}
const UserSignal = createSignal<User>(...)
```

```yaml
workflow:
  version: 1
  signals:
    results: UserSignal
  frames:
    - frame:
      view:
        component: OTUqiDataProvider
        props:
          source: trino-demo
          query: SELECT id, name FROM users
          results: :results
          output:
            results:
              - name: string
                id: number
```

If we wanted to also query for a user's email address or whatever, we'd have to either use a less defined signal (like the Records you've added in here) or also update the code definition of `UserSignal`.

I think my preferred way of doing this is what I said to you over tiki drinks the other day 🍹 What if we could declare the signal in the config?

```yaml
workflow:
  version: 1
  signals:
    results:
      - name: string
      - id: number
      - email: string
```

This idea is not fully baked :)

Reply:
Interesting! So the type definition is defined in the signal and the data component formats the results to conform with that? e.g.

```yaml
workflow:
  version: 1
  signals:
    id: NumberSignal
    results: UserSignal
    results2:
      - id: number
        name: string
  frames:
    - frame:
      view:
        component: OTUqiDataProvider
        props:
          source: trino-demo
          query: SELECT id, name FROM users
          outputs: [:id, :results, :results2]
```

I think I like this a lot. It declares the schema of the signal once instead of having an unnecessary abstraction in Record. This also feels the most terse yet expressive. The one slight downside is it moves the declaration of query results away from the query component, but I think this is fine because 1. this hopefully will never be hand written and 2. it should be clear that the outputs are :symbols and make it obvious to look at the signal store.

Reply:
Yeah, I like it too 👍

Reply:
Great. I added the following as a todo to implement this.

> Update to use new signal annotation https://github.com/open-truss/open-truss/pull/128/files#r1518165179


I am going to clean up this PR, merge it then start working on this new annotation.